channels_redis/core.py

```diff
@@ -130,7 +130,8 @@ def create_pool(self, index):
         host = self.hosts[index]
 
         if "address" in host:
-            return aioredis.ConnectionPool.from_url(host["address"])
+            address = host.pop("address")
```

Comment:
For this to work, we shouldn't mutate the original dict. We need to `deepcopy(host)`, and mutate that here.

Reply:
Also see #341 which addresses the same mutation issue for the branch below.

It might make sense to just `copy.deepcopy()` on line 130 to avoid the risk of mutation in the downstream `redis` library too.

Reply:
This should also be addressed in the pubsub layer (note that pubsub already performs a _copy_ of host dict in `RedisSingleShardConnection` init):

https://github.com/django/channels_redis/blob/a7094c58a15cbf6e621b2129253060fc80cfdfad/channels_redis/pubsub.py#L351